hs_core/views/utils.py

```diff
@@ -404,3 +404,251 @@ def show_relations_section(res_obj):
     if all_relation_count > has_part_count:
         return True
     return False
+
+
+def link_irods_file_to_django(resource, filename, size=0):
+    # link the newly created zip file to Django resource model
+    if resource:
```

Comment:
Yes, this function is called recursively for folder hierarchy, and it is always a good thing to check to make sure operations are only done for input object is not empty.